### PR TITLE
Correct minor typos

### PR DESCRIPTION
**Pull request by joshbeard**
### What

Two word corrections
### Why

They were grammatically incorrect
### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any. N/A

#### Content (n/a)
- [x] Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages. N/A
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

**Review by mgarrell777 — APPROVED** [review]
@joshbeard Thanks very much for your effort on this PR. I have a couple of small wording change suggestions. There's also a clarification I asked about. But on the whole, looks good.

File: website/docs/plugin/hashicorp-provider-design-principles.mdx [hunk 1]
```diff
@@ -41,7 +41,7 @@ The benefits of this practice include:
 Notes:
 
 - Dates and times should always be modeled in RFC 3339 whenever possible
-- Terraform does not support recursive types (but this can in some cases by modeled using a dynamic type)
+- Terraform does not support recursive types (but this can in some cases be modeled using a dynamic type)
 - Boolean attributes should be oriented so that true means to do something and false means not to do it, sometimes this means inverting the API
```
[the remark this answers is not in the document]
Comment (joshbeard) [reply website/docs/plugin/hashicorp-provider-design-principles.mdx:45]: I'm not completely clear on that one either, I guess. I think the former, though.

**Comment by joshbeard** [comment]
> @joshbeard Thanks very much for your effort on this PR. I have a couple of small wording change suggestions. There's also a clarification I asked about. But on the whole, looks good.

Thanks, @mgarrell777. I did think the wording could be improved but didn't take the liberty in doing that for the typo fix. I could use confirmation on that point of clarification, too.

File: website/docs/plugin/hashicorp-provider-design-principles.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgarrell777) [reply website/docs/plugin/hashicorp-provider-design-principles.mdx]: Ok, well I'll leave it as is for now and see what the Terraform devs say. Thanks again for your contribution.

**Comment by mgarrell777** [comment]
@joshbeard Were you able to get the clarification you needed on this PR? Or is that question still open?

**Pull request closed without merging**